.github/workflows/test-framework.yml

```diff
@@ -0,0 +1,42 @@
+name: run framework tests
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!master'
```

Comment:
If we have tests run with master, wouldn't we be more likely to catch problems earlier?

Reply:
I think tests will run on the PR but on on master after the PR has been merged

Reply:
Test will run on all branches except master. Therefore the !

Reply:
Yes I understand that. My question was, why shouldn't the tests run on master?

Reply:
ok. 2 reasons:
  - it's tested through the pull request before therefore not neccsessary
  - it's time consuming and prevents us for mass merges.
  -

Reply:
Thanks